src/overlays/actors/ovl_En_Sc_Ruppe/z_en_sc_ruppe.c

```diff
@@ -31,29 +37,155 @@ const ActorInit En_Sc_Ruppe_InitVars = {
+                return true;
+            }
+            break;
+        case 5:
+            if ((gSaveContext.save.weekEventReg[54] & 8)) {
+                gSaveContext.save.weekEventReg[54] &= (u8)~8;
+                return true;
+            }
+            break;
+    }
+    return false;
+}
+
+void func_80BD6A8C(EnScRuppe* this, GlobalContext* globalCtx) {
+    if (this->collider.base.ocFlags1 & OC1_HIT) {
+        this->rupeeDisplayTime = 0;
+        this->actor.gravity = 0.0f;
+        Actor_PlaySfxAtPos(&this->actor, NA_SE_SY_GET_RUPY);
+        func_801159EC(sRupeeInfo[this->rupeeIndex].amount);
+        this->actionFunc = func_80BD6B18;
+    }
+    this->actor.shape.rot.y += 0x1F4;
+    Actor_MoveWithGravity(&this->actor);
+}
+
+void func_80BD6B18(EnScRuppe* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    if (this->rupeeDisplayTime > 30) {
+        if (func_80BD697C(this->rupeeIndex)) {
+            Actor_MarkForDeath(&this->actor);
+        }
+    } else {
+        f32 scale;
+
+        this->rupeeDisplayTime++;
+        this->actor.world.pos = player->actor.world.pos;
+        this->actor.world.pos.y += 40.0f;
+        scale = (30.0f - this->rupeeDisplayTime) * 0.001f;
+        Actor_SetScale(&this->actor, scale);
+    }
+    this->actor.shape.rot.y += 0x3E8;
+    Actor_MoveWithGravity(&this->actor);
+}
+
+void EnScRuppe_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnScRuppe* this = THIS;
+    ColliderCylinder* collider = &this->collider;
+
+    Collider_InitCylinder(globalCtx, collider);
+    Collider_InitAndSetCylinder(globalCtx, collider, &this->actor, &sCylinderInit);
+    Actor_SetScale(&this->actor, 0.03f);
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 10.0f);
+    this->rupeeIndex = SCRUPPE_GET_PARAMS(thisx);
+    if ((this->rupeeIndex < 0) || (this->rupeeIndex >= 5)) {
+        this->rupeeIndex = 0;
+    }
+    this->actor.speedXZ = 0.0f;
+    this->actionFunc = func_80BD6A8C;
+    this->actor.gravity = -0.5f;
+    this->actor.shape.yOffset = 700.0f;
+}
+
+void EnScRuppe_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnScRuppe* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnScRuppe_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnScRuppe* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+    func_80BD6910(this, globalCtx);
+}
+
+void EnScRuppe_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    Gfx* opa;
+    EnScRuppe* this = THIS;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+    func_8012C28C(globalCtx->state.gfxCtx);
+    func_800B8050(&this->actor, globalCtx, 0);
+    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPSegment(POLY_OPA_DISP++, 0x08, Lib_SegmentedToVirtual(sRupeeInfo[this->rupeeIndex].tex));
+    gSPDisplayList(POLY_OPA_DISP++, gameplay_keep_DL_0622C0);
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
```

Comment:
```suggestion
    OPEN_DISPS(globalCtx->state.gfxCtx);

    func_8012C28C(globalCtx->state.gfxCtx);
    func_800B8050(&this->actor, globalCtx, 0);
    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
    gSPSegment(POLY_OPA_DISP++, 0x08, Lib_SegmentedToVirtual(sRupeeInfo[this->rupeeIndex].tex));
    gSPDisplayList(POLY_OPA_DISP++, gameplay_keep_DL_0622C0);

    CLOSE_DISPS(globalCtx->state.gfxCtx);
```
makes it easier to read. You can also name the symbols in the gameplay_keep xml for the displaylist and the textures.